instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/Instrumenter.java

```diff
@@ -193,12 +190,12 @@ private Context doStart(Context parentContext, REQUEST request, @Nullable Instan
     Span span = spanBuilder.setParent(context).startSpan();
     context = context.with(span);
 
-    if (!operationListeners.isEmpty()) {
+    if (operationListeners.length != 0) {
       // operation listeners run after span start, so that they have access to the current span
       // for capturing exemplars
       long startNanos = getNanos(startTime);
-      for (OperationListener operationListener : operationListeners) {
-        context = operationListener.onStart(context, attributes, startNanos);
+      for (int i = 0; i < operationListeners.length; i++) {
```

Comment:
could leave this as `for (OperationListener operationListener : operationListeners) {`